ecp5/pack.cc

```diff
@@ -2706,6 +2721,86 @@ class Ecp5Packer
                     copy_constraint(ci, id_CLK1, id_ECSOUT, 1);
                 } else if (ci->type == id_DCCA) {
                     copy_constraint(ci, id_CLKI, id_CLKO, 1);
+                } else if (ci->type == id_DCSC) {
+                    if ((!ci->ports.count(id_CLK0) && !ci->ports.count(id_CLK1)) || !ci->ports.count(id_DCSOUT))
+                        continue;
+                    auto mode = str_or_default(ci->params, id_DCSMODE, "POS");
+
+                    // TODO: We can do this simplification if we know
+                    // MODESEL is tied to 0. I don't know how I can check this
```

Comment:
to detect "tied to 0" or "tied to 1" at this stage, you can see if the connected net is called `$PACKER_GND_NET` or `"$PACKER_VCC_NET` accordingly.

Reply:
Fixed